### PR TITLE
Add preview logic

### PR DESCRIPTION
**Pull request by lpelizzaMS**
### Backlog

Add preview logic

### Changes
Created campaign preview view model and view.
Changed campaign model.
Created Link model.
Added new test.
Added a new method in campaign services.

## OLD
dopplercampaignpreview
## NEW
dopplercampaignpreviewhtml

### CODE REVIEW

File: Doppler.Mobile/Doppler.Mobile/Doppler.Mobile.Core/Services/CampaignService.cs [hunk 1]
```diff
@@ -20,20 +19,30 @@ public CampaignService(ILocalSettings localSettings, IDopplerAPI dopplerApi)
...
             if (!fetchCampaigns.IsSuccessResult)
                 return new Result<Page<Campaign>, string>(errorValue: fetchCampaigns.ErrorValue);
 
             var newPage = Mapper.Mapper.ToPageCampaign(fetchCampaigns.SuccessValue);
             return new Result<Page<Campaign>, string>(successValue: newPage);
         }
 
+        public Result<string, string> GetCampaignPreview(Campaign campaign)
+        {
+            var linkPreview = campaign.Links.FirstOrDefault(l => l.Relation.Contains("get-campaign-preview"));
```
Comment (andresmoschini) [inline Doppler.Mobile/Doppler.Mobile/Doppler.Mobile.Core/Services/CampaignService.cs]: If I am not wrong, Diego N. talked about showing HTML for preview and not the renderized image. Maybe both are useful and we could deal with HTML in another step.

**Comment by andresmoschini** [comment]
Nice work @lpelizzaMS !

Was the image shown in the PR description provided by Doppler? (I am asking because I do not remember if it has a good resolution, etc)

File: Doppler.Mobile/Doppler.Mobile/Doppler.Mobile.Core/Services/CampaignService.cs (hunk 1 above)
Comment (lpelizzaMS) [reply Doppler.Mobile/Doppler.Mobile/Doppler.Mobile.Core/Services/CampaignService.cs]: You are right but the API doesn't have any endpoint to return the HTML, the only thing that I have found It was the image. 
If you know about any endpoint that could be useful in this case, I don't have any problem to change the code and use HTML.

Comment (andresmoschini) [reply Doppler.Mobile/Doppler.Mobile/Doppler.Mobile.Core/Services/CampaignService.cs]: To get the HTML you can use `GET /accounts/{accountName}/campaigns/{campaignId}/content`

**Pull request merged**